src/rulesets/oas/functions/oasUnusedComponent.ts

```diff
@@ -0,0 +1,43 @@
+import type { IFunction, IFunctionContext, IFunctionResult } from '../../../types';
+
+export const oasUnusedComponent: IFunction<{}> = function (
+  this: IFunctionContext,
+  targetVal,
+  opts,
+  _paths,
+  otherValues,
+) {
+  const results: IFunctionResult[] = [];
+
+  if (targetVal.components === void 0) {
+    return results;
+  }
+
+  const componentTypes = [
+    'schemas',
+    'responses',
+    'parameters',
+    'examples',
+    'requestBodies',
+    'headers',
+    'links',
+    'callbacks',
+  ];
+
+  componentTypes.forEach(type => {
+    const resultsForType = this.functions.unreferencedReusableObject.call(
+      this,
+      targetVal.components[type],
+      { reusableObjectsLocation: `#/components/${type}` },
+      _paths,
+      otherValues,
+    );
+    if (resultsForType !== void 0 && Array.isArray(resultsForType)) {
+      results.push(...resultsForType);
+    }
+  });
```

Comment:
```suggestion
  for (const type of componentTypes) {
    const resultsForType = this.functions.unreferencedReusableObject.call(
      this,
      targetVal.components[type],
      { reusableObjectsLocation: `#/components/${type}` },
      _paths,
      otherValues,
    );
    if (resultsForType !== void 0 && Array.isArray(resultsForType)) {
      results.push(...resultsForType);
    }
  }
```

a very minor nitpick